lib/ecto/postgres/pg_framerate_migrations.ex

```diff
@@ -213,20 +263,72 @@ defpgmodule Vtc.Ecto.Postgres.PgFramerate.Migrations do
     ntsc_tags =
       Migration.constraint(
         table,
-        "#{field_name}_ntsc_tags",
+        "#{field}_ntsc_tags",
         check: """
         NOT (
-          ((#{field_name}).tags) @> '{drop}'::framerate_tags[]
-          AND ((#{field_name}).tags) @> '{non_drop}'::framerate_tags[]
+          ((#{target_value}).tags) @> '{drop}'::framerate_tags[]
+          AND ((#{target_value}).tags) @> '{non_drop}'::framerate_tags[]
         )
         """
       )
 
     %Constraint{} = Migration.create(ntsc_tags)
 
+    ntsc_valid =
+      Migration.constraint(
+        table,
+        "#{field}_ntsc_valid",
+        check: """
+        NOT #{function(:is_ntsc, Migration.repo())}(#{target_value})
+        OR (
+            (
+              round((#{target_value}).playback.numerator::float / (#{target_value}).playback.denominator::float) * 1000,
+              1001
+            )::rational
+            = (#{target_value}).playback
+        )
+        """
+      )
+
+    %Constraint{} = Migration.create(ntsc_valid)
+
+    drop_valid =
+      Migration.constraint(
+        table,
+        "#{field}_ntsc_drop_valid",
+        check: """
+        NOT (#{target_value}).tags @> '{drop}'::framerate_tags[]
+        OR (#{target_value}).playback % (30000, 1001)::rational = (0, 1)::rational
+        """
+      )
+
+    %Constraint{} = Migration.create(drop_valid)
+
     :ok
   end
 
+  @spec constraint_get_field_name(atom() | String.t(), atom() | String.t()) :: atom() | String.t()
+  defp constraint_get_field_name(target, nil), do: target
+  defp constraint_get_field_name(_, field_name), do: field_name
+
+  @doc """
+  Returns the config-qualified name of the function for this type.
+  """
+  @spec function(atom(), Ecto.Repo.t()) :: String.t()
+  def function(name, repo), do: "#{function_prefix(repo)}#{name}"
+
+  @spec function_prefix(Ecto.Repo.t()) :: String.t()
+  defp function_prefix(repo) do
+    functions_schema = get_config(repo, :functions_schema, :public)
+    functions_prefix = get_config(repo, :functions_prefix, "")
+
+    functions_prefix = if functions_prefix == "" and functions_schema == :public, do: "rational", else: functions_prefix
+
+    functions_prefix = if functions_prefix == "", do: "", else: "#{functions_prefix}_"
```

Comment:
you could probably refactor this to use a `cond...do` instead of multiple `if` statements

Reply:
In 1daac1c0020e0fe5ad4351659198cdc2c27c7255

I took the opportunity to reduce some code duplication and push the basic logic for this into the `Postgres.Utils` module.